File: web/src/types/inventory.ts
export interface Inventory {
  id: number
  warehouse_id: number
  product_id: number
  location_id: number | null
  quantity: number
  batch_no: string
  created_at: string
  updated_at: string
}

export interface CreateInventoryRequest {
  warehouse_id: number
  product_id: number
  location_id?: number
  quantity?: number
  batch_no?: string
}

export interface UpdateInventoryRequest {
  warehouse_id?: number
  product_id?: number
  location_id?: number
  quantity?: number
  batch_no?: string
}

export interface AdjustQuantityRequest {
  inventory_id: number
  quantity: number
}

File: internal/model/inventory.go
package model

type Inventory struct {
	BaseModel
	WarehouseID int64   `bun:"warehouse_id,notnull" json:"warehouse_id"`
	ProductID   int64   `bun:"product_id,notnull" json:"product_id"`
	LocationID  int64   `bun:"location_id" json:"location_id"`
	Quantity    float64 `bun:"quantity,notnull" json:"quantity"`
	BatchNo     string  `bun:"batch_no" json:"batch_no"`
}

func (i *Inventory) TableName() string {
	return "inventories"
}
